### PR TITLE
Pxt 381 multimodal rag

### PR DESCRIPTION
**Pull request by jacobweiss2305**
Add multimodal rag chat to our examples:

- AWS deployment template
- uv fastapi
- next js react frontend

### CODE REVIEW

**Comment by aaron-siegel** [comment]
This change is [Reviewable](https://reviewable.io/reviews/pixeltable/pixeltable/413)

File: docs/sample-apps/multimodal-chat/README.md [hunk 1]
```diff
@@ -0,0 +1,171 @@
+# Multimodal Chat Application
+
+A full-stack application demonstrating Pixel Table's multimodal processing capabilities. This application allows users to interact with documents, videos, and audio files using natural language queries.
```
Comment (pierrebrunelle) [inline docs/sample-apps/multimodal-chat/README.md:3]: Pixeltable

File: docs/sample-apps/multimodal-chat/README.md [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+```
+
+## Features
+
+- **Multimodal Data Support**: Process and analyze documents, videos, and audio files
+- **Advanced Processing**: Leverages [Pixeltable's data types and formats](https://docs.pixeltable.com/docs/data-types-and-formats)
+- **Flexible LLM Integration**: 
+  - Default: [OpenAI](https://docs.pixeltable.com/docs/working-with-openai)
+  - Alternative options: [Ollama](https://docs.pixeltable.com/docs/working-with-ollama), [LLama.cpp](https://docs.pixeltable.com/docs/working-with-llamacpp)
+- **Local Processing Options**: Use [Whisper/WhisperX](https://docs.pixeltable.com/docs/whisper) for local transcription
```
Comment (pierrebrunelle) [inline docs/sample-apps/multimodal-chat/README.md:51]: make it clear that for now it uses OpenAI Whisper but yes can use Whisper/WhisperX instead

File: docs/sample-apps/multimodal-chat/backend/api/routes.py [hunk 3]
```diff
@@ -0,0 +1,583 @@
...
+        "\n\n---\n\n".join(context_parts)
+        if context_parts
+        else "No relevant context found."
+    )
+
+    return f"Context:\n{full_context}\n\nQuestion:\n{question}"
+
+
+# Add computed columns for video processing
+docs_table["audio_extract"] = extract_audio(docs_table.video, format="mp3")
```
Comment (pierrebrunelle) [inline docs/sample-apps/multimodal-chat/backend/api/routes.py]: change all the computed column to use:

**.add_computed_column(name**=......)**)**

File: docs/sample-apps/multimodal-chat/frontend/components/ChatInterface.tsx [hunk 4]
```diff
@@ -0,0 +1,883 @@
...
+  useEffect(() => {
+    messagesEndRef.current?.scrollIntoView({ behavior: 'smooth' });
+  }, [messages]);
+
+  useEffect(() => {
+    fetchFiles();
+  }, []);
+
+
+  // Utility functions
```
Comment (pierrebrunelle) [inline docs/sample-apps/multimodal-chat/frontend/components/ChatInterface.tsx]: validateFile is not used anymore. Dead code. Can remove.

File: docs/sample-apps/multimodal-chat/backend/api/routes.py [hunk 5]
```diff
@@ -0,0 +1,583 @@
...
+        docs_table = pxt.get_table("chatbot.documents")
+        docs_table.insert([{"document": abs_path}])
+        return True
+
+    except Exception as e:
+        logger.error(f"Error processing document: {e}")
+        raise
+
+
+@router.post("/api/upload")
```
Comment (pierrebrunelle) [inline docs/sample-apps/multimodal-chat/backend/api/routes.py:243]: let me come back to you on that one as we should (1) remove the check for video file that is superfluous now that we have the other api/videos/upload and want to see if I can get the .py -> .md working if not we can also ditch that code.

**Comment by pierrebrunelle** [comment]
We made the appropriate changes. @aaron-siegel feel free to review it.

**Comment by jacobweiss2305** [comment]
Closing in favor of my fork:
https://github.com/pixeltable/pixeltable/pull/419

**Pull request closed without merging**